src/modules/rolle/api/rolle.controller.ts

```diff
@@ -274,27 +274,41 @@ export class RolleController {
                 ),
             );
         }
-        const result: void | DomainError = await rolle.attachServiceProvider(spBodyParams.serviceProviderId);
+        const result: void | DomainError = await rolle.updateServiceProviders(spBodyParams.serviceProviderIds);
         if (result instanceof DomainError) {
             throw SchulConnexErrorMapper.mapSchulConnexErrorToHttpException(
                 SchulConnexErrorMapper.mapDomainErrorToSchulConnexError(result),
             );
         }
         await this.rolleRepo.save(rolle);
-        const serviceProvider: Option<ServiceProvider<true>> = await this.serviceProviderRepo.findById(
-            spBodyParams.serviceProviderId,
+
+        const serviceProviderMap: Map<string, ServiceProvider<true>> = await this.serviceProviderRepo.findByIds(
+            spBodyParams.serviceProviderIds,
+        );
+
+        // Check if all provided IDs are in the map
+        const missingServiceProviderIds: string[] = spBodyParams.serviceProviderIds.filter(
+            (id: string) => !serviceProviderMap.has(id),
         );
-        if (!serviceProvider) {
+
+        if (missingServiceProviderIds.length > 0) {
+            // If some IDs are missing, throw an error with details about the missing IDs
             throw SchulConnexErrorMapper.mapSchulConnexErrorToHttpException(
                 new SchulConnexError({
                     code: 500,
                     subcode: '00',
                     titel: 'Service-Provider nicht gefunden',
-                    beschreibung: 'Der Service-Provider konnte nach Zuweisung zur Rolle nicht gefunden werden!',
+                    beschreibung: `Die folgenden Service-Provider-IDs konnten nicht gefunden werden: ${missingServiceProviderIds.join(', ')}`,
                 }),
             );
         }
-        return new ServiceProviderResponse(serviceProvider);
+        // Convert the Map of service providers to an array of ServiceProviderResponse objects
+        const serviceProviderResponses: ServiceProviderResponse[] = Array.from(serviceProviderMap.values()).map(
+            (serviceProvider: ServiceProvider<true>) => new ServiceProviderResponse(serviceProvider),
+        );
+
+        // Return the array of ServiceProviderResponse objects
+        return serviceProviderResponses;
     }
 
     @Delete(':rolleId/serviceProviders')
```

Comment:
If we only ever send the entire list of serviceproviders, do we still need this endpoint?

Reply:
I asked for it to be removed but Kristoff wanted it to stay so I left it.